### PR TITLE
Disallow admins from linking LTI accounts

### PR DESCRIPTION
**Pull request by nicklathe**
Adds logic to disallow linking for admins, and returns an error message to them. This alert is displayed after they attempt to link the account, as that's the point where the existing user account is looked up and determined to be an admin.

Since this error message will only be for internal Code.org users, a simple flash alert was used.

**Alert**
Screenshot 2024-06-26 at 9 00 24 AM

## PR Checklist:

- [ ] Tests provide adequate coverage
- [ ] Privacy and Security impacts have been assessed
- [ ] Code is well-commented
- [ ] New features are translatable or updates will not break translations
- [ ] Relevant documentation has been added or updated
- [ ] User impact is well-understood and desirable
- [ ] Pull Request is labeled appropriately
- [ ] Follow-up work items (including potential tech debt) are tracked and linked

### CODE REVIEW

File: dashboard/app/controllers/lti/v1/account_linking_controller.rb [hunk 1]
```diff
@@ -31,6 +31,10 @@ def link_email
         head :bad_request unless PartialRegistration.in_progress?(session)
         params.require([:email, :password])
         existing_user = User.find_by_email_or_hashed_email(params[:email])
+        if existing_user.admin?
+          flash[:alert] = "Admins cannot link LTI accounts"
```
Comment (nicklathe) [inline dashboard/app/controllers/lti/v1/account_linking_controller.rb]: TODO: Add this string to i18n JSON

File: dashboard/test/controllers/lti/v1/account_linking_controller_test.rb [hunk 2]
```diff
@@ -3,6 +3,7 @@
 class Lti::V1::AccountLinkingControllerTest < ActionController::TestCase
   setup do
     @user = create(:teacher, email: 'test@lti.com')
+    @admin = create :admin, email: 'admin@lti.com'
```
Comment (nicklathe) [inline dashboard/test/controllers/lti/v1/account_linking_controller_test.rb]: Interestingly, `@admin.admin` here is `true` but this test fails, and when this user is looked up in the link_email endpoint, it's nil.

File: dashboard/app/controllers/omniauth_callbacks_controller.rb [hunk 3]
```diff
@@ -549,6 +549,10 @@ def login
   # For linking new LTI auth options to existing accounts
   private def link_accounts(user)
     if user
+      if user.admin
```
Comment (carl-codeorg) [inline dashboard/app/controllers/omniauth_callbacks_controller.rb]: Any reason this check is `admin` sans `?`?

**Review by wilkie — APPROVED** [review]
LGTM

File: dashboard/app/controllers/omniauth_callbacks_controller.rb (hunk 3 above)
Comment (nicklathe) [reply dashboard/app/controllers/omniauth_callbacks_controller.rb]: Yeah, good catch. I meant to add the `?` in for this.

**Review by carl-codeorg — APPROVED** [review]

**Pull request merged**